packages/cli/src/TagHelpers.ts

```diff
@@ -1,109 +1,128 @@
+	const result = await getConnection().createQueryBuilder()
+		.select()
+		.from('workflows_tags', 'workflows_tags')
+		.where('workflowId = :workflowId AND tagId = :tagId', { workflowId, tagId })
+		.execute();
 
-	if (tag) {
-		throw new ResponseHelper.ResponseError('Tag name already exists.', undefined, 400);
-	}
+	return result.length > 0;
 }
 
 /**
- * Validate that a tag and a workflow are not related so that a link can be created.
+ * Check whether a tag ID exists in the `tag_entity` table.
+ *
+ * Used for creating a workflow or updating a tag.
  */
-export async function validateNoRelation(workflowId: number, tagId: number): Promise<void> | never {
-	const areRelated = await checkRelated(workflowId, tagId);
+export async function exists(id: string): Promise<void> | never {
+	const tag = await Db.collections.Tag!.findOne({ where: { id }});
 
-	if (areRelated) {
-		throw new ResponseHelper.ResponseError(`Workflow ID ${workflowId} and tag ID ${tagId} are already related.`, undefined, 400);
+	if (!tag) {
+		throw new ResponseHelper.ResponseError(`Tag with ID ${id} does not exist.`, undefined, 400);
 	}
 }
 
+// ----------------------------------
+//           validators
+// ----------------------------------
+
 /**
- * Validate that a tag and a workflow are related so that their link can be deleted.
+ * Validate whether every tag ID is a string array and exists in the `tag_entity` table.
+ *
+ * Used for creating a workflow or updating a tag.
  */
-export async function validateRelation(workflowId: number, tagId: number): Promise<void> | never {
-	const areRelated = await checkRelated(workflowId, tagId);
+export async function validateTags(tags: unknown[]): Promise<void> | never {
+	if (!isStringArray(tags)) {
+		throw new ResponseHelper.ResponseError(`The tags property is not an array of strings.`, undefined, 400);
+	}
 
-	if (!areRelated) {
-		throw new ResponseHelper.ResponseError(`Workflow ID ${workflowId} and tag ID ${tagId} are not related.`, undefined, 400);
+	for (const tagId of tags) {
+		await exists(tagId);
 	}
 }
 
 /**
- * Validate whether the request body for a create/update operation has a `name` property.
+ * Validate whether a tag name
+ * - is present in the request body,
+ * - is a string,
+ * - is 1 to 24 characters long, and
+ * - does not exist already.
+ *
+ * Used for creating or updating a tag.
  */
-export function validateRequestBody({ name }: { name: string | undefined }): void | never {
-	if (!name) {
+export async function validateName(name: unknown): Promise<void> | never {
+	if (name === undefined) {
 		throw new ResponseHelper.ResponseError(`Property 'name' missing from request body.`, undefined, 400);
 	}
-}
 
+	if (typeof name !== 'string') {
+		throw new ResponseHelper.ResponseError(`Property 'name' must be a string.`, undefined, 400);
+	}
 
-// ----------------------------------
-//             queries
-// ----------------------------------
+	if (name.length <= 0 || name.length > 24) {
```

Comment:
24 is magic number - so maybe add a variable to top of file or some other configuration file

```
const TAG_NAME_LENGTH_LIMIT = 24;
```

```suggestion
	if (name.length <= 0 || name.length > TAG_NAME_LENGTH_LIMIT) {
```